interoperator/controllers/provisioners/sfserviceinstancecleaner/sfserviceinstancecleaner_controller_test.go

```diff
@@ -0,0 +1,129 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	osbv1alpha1 "github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/api/osb/v1alpha1"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/constants"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var binding = &osbv1alpha1.SFServiceBinding{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "binding-id",
+		Namespace: "default",
+		Labels: map[string]string{
+			"state": "in_queue",
+		},
+		Finalizers: []string{constants.BrokerFinalizer},
+	},
+	Spec: osbv1alpha1.SFServiceBindingSpec{
+		ID:                "binding-id",
+		InstanceID:        "instance-id",
+		PlanID:            "plan-id",
+		ServiceID:         "service-id",
+		AcceptsIncomplete: true,
+	},
+	Status: osbv1alpha1.SFServiceBindingStatus{
+		State: "in_queue",
+	},
+}
+
+var c client.Client
+
+const timeout = time.Second * 2
+
+func setupInteroperatorConfig(g *gomega.GomegaWithT) {
+	data := make(map[string]string)
+	data["instanceWorkerCount"] = "1"
+	data["bindingWorkerCount"] = "1"
+	watchList := `
+- apiVersion: kubedb.com/v1alpha1
+  kind: Postgres
+- apiVersion: kubernetes.sapcloud.io/v1alpha1
+  kind: Postgresql
+- apiVersion: deployment.servicefabrik.io/v1alpha1
+  kind: Director`
+	data["instanceContollerWatchList"] = watchList
+	data["bindingContollerWatchList"] = watchList
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.ConfigMapName,
+			Namespace: constants.DefaultServiceFabrikNamespace,
+		},
+		Data: data,
+	}
+	g.Expect(c.Create(context.TODO(), configMap)).NotTo(gomega.HaveOccurred())
+}
+
+func TestReconcile(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	mgr, err := manager.New(cfg, manager.Options{
+		MetricsBindAddress: "0",
+	})
+	g.Expect(err).NotTo(gomega.HaveOccurred())
+	c = mgr.GetClient()
```

Comment:
It safer to create a new client from cfg for tests. Otherwise the tests can fail intermittently due to cache stuff

Reply:
Done